src/functions-and-arrays.js

```diff
@@ -1,41 +1,116 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(number1, number2) {
+  if (number1 > number2) {
+    return number1;
+  } else if (number1 < number2) {
+    return number2;
+  }else if (number1 === number2)
+  return number1
+  }
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(arr) {
+if (arr.length === 0) return null;
+let longestWord = ""
+for(let i = 0; i < arr.length; i++) {
+   if(arr[i].length > longestWord.length){
+    longestWord = arr[i]
+  }
+}
+return longestWord;
+}
 
+console.log(findLongestWord(words))
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numbers) {
+  let sum = 0;
+  numbers.forEach((number) => {
+      // vai somar cada número da array ao resultado a soma anterior
+      sum = sum + number;
+  })
+  return sum;
+}
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(arr) {
+  let sum = 0;
+  for (i=0; i < arr.length; i++) {
+    if (typeof arr[i] === "number") {
+      sum += arr[i];
+    } else if (typeof arr[i] === "string") {
+      sum += arr[i].length;
+    } else if (typeof arr[i] === "boolean") {
+      if (arr[i] === true) {
+        sum += 1;
+      }
+    }
+   }
+   return sum;
+}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(arrNum) {
+  let sum = 0;
+  if (arrNum.length === 0){
+    return null;
+  }
+  for (let i = 0; i < arrNum.length; i++){
+    sum = sum + arrNum[i]
```

Comment:
You already did it in the previous iteration, let's do it in this one too:

`sum += arrNum[i];`

Also, try to be consistent with your use of `;`